### PR TITLE
add Iframe component

### PR DESCRIPTION
**Pull request by overra**
Got a start on this, not sure if there's a better way to reuse the `iframe` node instance or if it's even necessary. Might need to convert to class component to remove the node on unmount.

### CODE REVIEW

File: src/Iframe.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+import { cache } from "./cache";
+
+function load(attributes) {
+  const { src, height, ...attrs } = attributes;
+  return new Promise((resolve, reject) => {
+    const iframe = document.createElement("iframe");
+    Object.keys(attrs).forEach(name => iframe.setAttribute(name, attrs[name]));
+    iframe.src = src;
+    iframe.addEventListener(
+      "load",
```
Comment (jaredpalmer) [inline src/Iframe.js]: can you just use `onLoad`?

Comment (overra) [reply src/Iframe.js]: sure

File: src/Iframe.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    iframe.height = 0;
+    document.body.appendChild(iframe);
+  });
+}
+
+const resource = createResource(load, ({ href }) => href);
+
+export const Iframe = props => {
+  const iframeNode = resource.read(cache, props);
+  return <iframe ref={ref => ref.replaceWith(iframeNode)} {...props} />;
```
Comment (crosscompile) [inline src/Iframe.js]: We might need to use something other than `replaceWith`, there are [some gaps in browser support](https://developer.mozilla.org/en-US/docs/Web/API/ChildNode/replaceWith#Browser_compatibility).  We'll also need to handle the case where `ref` is null.

Comment (overra) [reply src/Iframe.js]: Turns out this wasn't necessary.

**Comment by crosscompile** [comment]
Thinking about this more, I'm not sure how to handle an `<Iframe />` component since an iframe only starts loading once it is added to the DOM and with suspense we don't want to commit anything to the DOM until it is ready.

Currently appending it to the body with height zero results in the iframe loading twice.  We could also try [preloading](https://developer.mozilla.org/en-US/docs/Web/HTML/Preloading_content) the document but it is not supported in all browsers.

Maybe if we made a `<Preload />` component that includes a [polyfilll](https://github.com/digitalkaoz/preload-polyfill) that would be more transparent about what browser elements are being used.

**Comment by overra** [comment]
Good point. Even with this `<Iframe />` it doesn't seem that `onload` is triggered when the contents are fully loaded. 

I'm not sure if I'm doing [this](https://codesandbox.io/s/o447okxmjq) wrong but it's not making a network request until the iframe is rendered.

**Pull request closed without merging**